nslsii/tests/temperature_controllers_test.py

```diff
@@ -44,6 +45,19 @@ def test_Eurotherm(RE):
         # check that the readback value is within euro.tolerance of 100
         assert abs(euro.readback.get() - 100) <= euro.tolerance.get()
 
+        # test that the set will fail after 'timeout'
+        euro.timeout.set(1)
+        with pytest.raises(FailedStatus):
+            RE(mv(euro,100))
+        # ensure callback is removed
+        assert len(euro.readback._callbacks['value']) == 0
+        euro.timeout.set(500) # reset to default for the following tests.
+
+        # test that the lock prevents setting while set in progress
+        with pytest.raises(Exception):
```

Comment:
This is another reason more narrow, case-specific exceptions are good to have. If the test inside that block starts failing for some _other_ reason than the expected one, `Exception` does not distinguish and the test will still pass.

Reply:
noted